services/elastic_ip_manager/src/main/java/com/futurewei/alcor/elasticipmanager/service/implement/ElasticIpServiceImpl.java

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.elasticipmanager.service.implement;
+
+import com.futurewei.alcor.elasticipmanager.dao.ElasticIpAllocator;
+import com.futurewei.alcor.elasticipmanager.dao.ElasticIpRangeRepo;
+import com.futurewei.alcor.elasticipmanager.dao.ElasticIpRepo;
+import com.futurewei.alcor.elasticipmanager.exception.elasticip.ElasticIpAllocationException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticip.ElasticIpInUseException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticip.ElasticIpNotFoundException;
+import com.futurewei.alcor.elasticipmanager.exception.ElasticIpQueryFormatException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticiprange.ElasticIpRangeInUseException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticiprange.ElasticIpRangeNotFoundException;
+import com.futurewei.alcor.elasticipmanager.service.ElasticIpService;
+import com.futurewei.alcor.web.entity.elasticip.ElasticIp;
+import com.futurewei.alcor.web.entity.elasticip.ElasticIpRange;
+import com.futurewei.alcor.web.entity.elasticip.ElasticIpInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.*;
+
+
+@Service
+public class ElasticIpServiceImpl implements ElasticIpService {
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticIpServiceImpl.class);
+
+    @Autowired
+    ElasticIpRepo elasticIpRepo;
+
+    @Autowired
+    ElasticIpRangeRepo elasticIpRangeRepo;
+
+    @Autowired
+    ElasticIpAllocator elasticIpAllocator;
+
+    public ElasticIpInfo createElasticIp(ElasticIpInfo request) throws Exception {
+        LOG.debug("Create an elastic ip, request: {}", request);
+
+        String rangeId = request.getRangeId();
+        ElasticIpRange range = elasticIpRangeRepo.findItem(rangeId);
```

Comment:
I think range id is not available to our end users.

Reply:
OK